src/Conversion/ONNXToKrnl/Tensor/ArgMinMax.cpp

```diff
@@ -19,27 +20,71 @@
 using namespace mlir;
 
 namespace onnx_mlir {
-struct ONNXArgMaxOpLowering : public ConversionPattern {
-  ONNXArgMaxOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
-      : ConversionPattern(
-            typeConverter, mlir::ONNXArgMaxOp::getOperationName(), 1, ctx) {}
+
+template <typename ArgOp>
+inline Value isNewVal(MultiDialectBuilder<KrnlBuilder, MathBuilder> create,
+    Value next, Value dstVal);
+
+template <>
+inline Value isNewVal<ONNXArgMinOp>(
+    MultiDialectBuilder<KrnlBuilder, MathBuilder> create, Value next,
+    Value dstVal) {
+  return create.math.slt(next, dstVal);
+}
+
+template <>
+inline Value isNewVal<ONNXArgMaxOp>(
+    MultiDialectBuilder<KrnlBuilder, MathBuilder> create, Value next,
+    Value dstVal) {
+  return create.math.sgt(next, dstVal);
+}
+
+template <typename ArgOp>
+inline llvm::SmallVector<IndexExpr, 4> getOutputDims(ArgOp *op,
+    typename ArgOp::Adaptor operandAdaptor, mlir::OpBuilder *rewriter,
+    ArrayValueIndexCapture::GetDenseVal fGetDenseVal,
+    ArrayValueIndexCapture::LoadVal fLoadVal);
+
+template <>
+inline llvm::SmallVector<IndexExpr, 4> getOutputDims<ONNXArgMinOp>(
+    ONNXArgMinOp *op, typename ONNXArgMinOp::Adaptor operandAdaptor,
+    mlir::OpBuilder *rewriter, ArrayValueIndexCapture::GetDenseVal fGetDenseVal,
+    ArrayValueIndexCapture::LoadVal fLoadVal) {
+  ONNXArgMinOpShapeHelper shapeHelper(op, rewriter, fGetDenseVal, fLoadVal);
+  auto shapecomputed = shapeHelper.computeShape(operandAdaptor);
+  (void)shapecomputed;
+  assert(!failed(shapecomputed) && "expected to succeed");
+  return shapeHelper.dimsForOutput();
+}
+
+template <>
+inline llvm::SmallVector<IndexExpr, 4> getOutputDims<ONNXArgMaxOp>(
+    ONNXArgMaxOp *op, typename ONNXArgMaxOp::Adaptor operandAdaptor,
+    mlir::OpBuilder *rewriter, ArrayValueIndexCapture::GetDenseVal fGetDenseVal,
+    ArrayValueIndexCapture::LoadVal fLoadVal) {
+  ONNXArgMaxOpShapeHelper shapeHelper(op, rewriter, fGetDenseVal, fLoadVal);
+  auto shapecomputed = shapeHelper.computeShape(operandAdaptor);
+  (void)shapecomputed;
+  assert(!failed(shapecomputed) && "expected to succeed");
+  return shapeHelper.dimsForOutput();
+}
+
+template <typename ArgOp>
+struct ONNXArgMinMaxOpLowering : public ConversionPattern {
+  ONNXArgMinMaxOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(typeConverter, ArgOp::getOperationName(), 1, ctx) {}
 
   LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
       ConversionPatternRewriter &rewriter) const final {
     // Gather info.
     auto loc = op->getLoc();
-    ONNXArgMaxOpAdaptor operandAdaptor(operands);
-    ONNXArgMaxOp argMaxOp = llvm::cast<ONNXArgMaxOp>(op);
+    ArgOp argOp = llvm::cast<ArgOp>(op);
 
```

Comment:
CI failed because of `onnx_mlir::IndexExprImpl::IndexExprImpl(): Assertion scope && "expected IndexExpr Scope to be defined"' failed.`. It seems we need to declare an IndexExpr scope at the beginning of this function, e.g. `IndexExprScope scope(&rewriter, loc)`.